### DISCUSSION TITLE
How to add `AddHeadersToRequest` globally?

### DISCUSSION THREAD

**Discussion by pradeep1403** [ThreeMammals/Ocelot#1992] [Q&A] [answered]
I am having the multiple Routes with common Headers in the `AddHeadersToRequest` property. It is difficult to add it in the individual route level.

My sample Route as below:
```json
{
  "DownstreamPathTemplate": "/api/inbound/{RestRoute}",
  "DownstreamScheme": "http"
  "DownstreamHostAndPorts": [
	{ "Host": "Localhost", "Port": 51712 }
  ],
  "AuthenticationOptions":{
	"AuthenticationProviderKey": "ApiKey",
	"AlLowedScopes": []
  },
  "UpstreamPathTemplate": "/api/dev/inbound/{RestRoute}",
  "UpstreamHttpMethod": [ "post","get", "Options"],
  "AddHeadersToRequest":{
      "JWT":"Claims[JWT]>value"
   }
}
```
How can we add the `AddHeadersToRequest` in the global level that means instead of writing the `AddHeadersToRequest` in the each route in the **ocelot.json** file?

**Comment by raman-m** [comment, marked as answer]
Hi @pradeep1403 !
As far as I know, global feature is not implemented. You need to define for each route.
I recommend to have less routes using [Catch All](https://ocelot.readthedocs.io/en/latest/features/routing.html#catch-all) feature, and global options will not be required.
